src/main/java/edu/wpi/grip/core/Palette.java

```diff
@@ -0,0 +1,47 @@
+package edu.wpi.grip.core;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import edu.wpi.grip.core.events.OperationAddedEvent;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.python.jline.internal.Preconditions.checkNotNull;
+
+/**
+ * The palette is a library of operations that can be added as steps in the {@link Pipeline}
+ */
+public class Palette {
+
+    private final EventBus eventBus;
+    private final Map<String, Operation> operations = new HashMap<>();
+
+    public Palette(EventBus eventBus) {
+        checkNotNull(eventBus);
+        this.eventBus = eventBus;
```

Comment:
This could be `this.eventBus = checkNotNull(eventBus, "The Event Bus can not be null");`